### DISCUSSION TITLE
Dishwasher  G7310 program start and door status issues

### DISCUSSION THREAD

**Discussion by Labu7** [astrandb/miele#217] [Q&A]
Hi all,

I have just deployed the integration for my dishwasher and found 2 issues:

1 the door status always report closed
2 the program start button is never available. After some indications from @astrandb I found that the stop was working fine (enabled when the program is running) but no way to have the start.

If I do the start from  the Miele Android App, it is taken in account and I can then stop from HA.

Herewith the status of the associated entities

Any ideas welcomed.

Thanks
![Dish1](https://user-images.githubusercontent.com/109757863/215091412-3057c6bd-2773-444d-8892-288822a596b8.JPG)
![Dish2](https://user-images.githubusercontent.com/109757863/215091448-60a18bc4-340e-492b-a210-31e83c0b2f7a.JPG)

**Comment by astrandb** [comment]
The integration just displays what the Miele API reports. You can check the reported door status in the diagnostic download files with the door open and closed. 
Miele uses one API for internal use in the apps and another API for public 3:rd party use. Unfortunately the public one is far from perfect. The start and stop buttons are enabled/disabled by the API and the API just send an error message if you try to override the reported available processActions.

**Reply by Labu7** [reply]
Good to know, from what I read there is limited support around the API. I will try some investigations on this side.

Great job from your side, may be the french translation could be my small contribution. It works for me.

**Reply by aturri** [reply]
Hi,
Start button is enabled only when you delay start the appliance (in order to start it earlier, manually). I'm not sure, but it could also be enabled if MobileStart button is pressed on device.

**Comment by imbrechtsjurgen** [comment]
hi @Labu7 and @astrandb, maybe you've already found out but here it goes.
The miele API only allows starting a program when you've put the dishwasher in "smartstart mode".
You should change the settings of the machine from "EcoStart" to "SmartStart".
(Ecostart is based on fixed times when energyprices are low, SmartStart is based on a trigger).

After enabling smartstart you can select a program on the machine, tap on the "delay start" button and select a maximum time when the machine should be finished and you close the machine.
This will put the machine in "waiting for start" and you are able to start it from home assistant.
I was just testing this and it works as expected.

I'm a little dissapointed that we cannot select programs/times/functions from the API but that's Miele's choice.

But the features can be used. I will explain how I envision them.
In the morning before work we fill up the last pieces before heading out. We select a desired program and select "SmartStart" from the "delayed start" menu. I select 18:00 as the maximum timestamp for the machine to be finished.
Throughout the day an automation in HomeAssistant will monitor the solar input (and optionally a home battery) what would be the best moment for starting the machine. I guess it would be best to trigger the start when my solar panels are producing 500W more then I'm using at that specific moment. I see that our machine uses an average of 400KWh during a cycle. Although I should measure this to know the peaks. 

If home assistant doesn't pull the trigger, the machine will start by itself to make sure the dishes are clean before the requested time.

You could also use this setup with a dynamic power contract. You could then plan to start the cycle when powerpricing is the lowest of the night/day.

FYI: my machine also has a "SolarSave" program. This program does not heat the water. This could potentially be used when you have a small boiler next to the machine that you can "heat up" with excess solar power. 
This means you could heat up the water in a small 10 or 15 liter boiler to 60°C or 70°C at a moment when there is excess solar power.
and for example, in the evening, when there is no more solar but you would like to do the dishes, you can run your machine.

I'm sorry for the long post but I wanted to output my thoughts and maybe help somebody with this.

**Comment by Isusami** [comment]
Hi all,
From what I read from the threads, I cannot start my washing machine by HA without delay start?
@astrandb

**Reply by astrandb** [reply]
Well, it depends...
First - there is no official documentation from Miele how the details are supposed to work.
Different machine models behave sligthly differently so you will have to experiment with your own apppliance.
I can start my washing machine (WCI870) from HA by:
1. Loading the machine and closing the door.
2. Turn the dial to "MobileStart".
3. Press "Start" on the touchscreen.
4. Wait for whatever you find appropriate or run it immediately.
5. Send a `Set program` action with device_id and program_id from HA. Goto Developer tools to create the command.

**Reply by Isusami** [reply]
I've received this error after `set program`

> Failed to perform the action miele.set_program. Service set_program: 400 Bad Request


Is this a bug, or does my dishwasher model not support the start program?
